cmd/cli/app/provider/provider_update.go

```diff
@@ -0,0 +1,509 @@
+	var structConfig map[string]any
+	bytes, err = json.Marshal(serde)
+	if err != nil {
+		// TODO this is likely to be an internal error and
+		// should be mapped to a more suitable user-facing
+		// error.
+		return cli.MessageAndError(
+			"invalid config",
+			err,
+		)
+	}
+	if err := json.Unmarshal(bytes, &structConfig); err != nil {
+		// TODO this is likely to be an internal error and
+		// should be mapped to a more suitable user-facing
+		// error.
+		return cli.MessageAndError(
+			"invalid configuration",
+			err,
+		)
+	}
+
+	cfg, err := structpb.NewStruct(structConfig)
+	if err != nil {
+		return cli.MessageAndError("invalid config patch", err)
+	}
+
+	req := &minderv1.PatchProviderRequest{
+		Context: &minderv1.Context{
+			Project:  &project,
+			Provider: &providerName,
+		},
+		Patch: &minderv1.Provider{
+			Config: cfg,
+		},
+	}
+
+	_, err = client.PatchProvider(ctx, req)
+	if err != nil {
+		return cli.MessageAndError("failed calling minder", err)
+	}
+
+	cmd.Println("Provider updated successfully")
+
+	return nil
+}
+
+func parseConfigAttribute(attr string) (string, string, error) {
+	parts := strings.SplitN(attr, "=", 2)
+	if len(parts) != 2 {
+		return "", "", fmt.Errorf("invalid attribute format: %s", attr)
+	}
+	if parts[0] == "" || parts[1] == "" {
+		return "", "", fmt.Errorf("invalid attribute format: %s", attr)
+	}
+	return parts[0], parts[1], nil
+}
+
+func configAttribute(
+	config reflect.Value,
+	attrName string,
+	attrValue *string,
+) error {
+	if config.Kind() != reflect.Pointer {
+		return errors.New("config must be passed by reference")
+	}
+
+	// Attribute name is meant to be a path traversing struct
+	// fields of the form <root>.<field1>.<field2>...
+	attrPath := strings.Split(attrName, ".")
+
+	if err := recurConfigAttribute(config, attrPath, attrValue); err != nil {
+		return fmt.Errorf("%s is not a valid attribute: %w", attrName, err)
+	}
+
+	return nil
+}
+
+func recurConfigAttribute(
+	config reflect.Value,
+	path []string,
+	attrValue *string,
+) error {
+	//nolint:exhaustive
+	switch config.Kind() {
+	case reflect.Pointer:
+		return recurConfigAttribute(
+			reflect.Indirect(config),
+			path, // we just dereference the pointer
+			attrValue,
+		)
+	case reflect.Struct,
+		reflect.Map:
+		if len(path) == 0 {
+			return errors.New("too short")
+		}
+
+		fd, err := next(config, path[0])
+		if err != nil {
+			return err
+		}
+		return recurConfigAttribute(
+			fd,
+			path[1:],
+			attrValue,
+		)
+	case reflect.Bool,
+		reflect.Int32,
+		reflect.Int64,
+		reflect.Uint32,
+		reflect.Uint64,
+		reflect.Float32,
+		reflect.Float64,
+		reflect.String:
+		if len(path) != 0 {
+			return errors.New("too long")
+		}
+		if attrValue != nil {
+			parserFunc, found := parserMap[config.Kind()]
+			if !found {
+				return fmt.Errorf("field has unexpected kind: %s", config.Kind())
+			}
+			v, err := parserFunc(*attrValue)
+			if err != nil {
+				return fmt.Errorf("expected bool, got %s", *attrValue)
```

Comment:
the literal bool in the error message is likely a typo?

Reply:
Yes, thanks for catching it.

Reply:
I changed it to this
```
$ minder provider update -j ... -n ... --set-attribute auto_registration.entities.repository.enabled=50
...
Message: invalid attribute
Details: auto_registration.entities.repository.enabled is not a valid attribute: expected bool, got 50
```